pctasks/core/tests/storage/test_blob.py

```diff
@@ -126,3 +126,10 @@ def test_blob_download_timeout():
 def test_maybe_rewrite_blob_storage_url(url, expected):
     result = maybe_rewrite_blob_storage_url(url)
     assert result == expected
+
+
+def test_generate_container_sas():
+    storage = BlobStorage("devstoraccount1", "test-data")
+    result = storage._generate_container_sas()
```

Comment:
This is currently failing with azurite.

Reply:
I might give up on this test. We need to somehow get a `credential` to `generate_container_sas` used for the signature. Previously, we only used SAS tokens we obtained from the account key (see `BlobStorage.from_accont_key)`. This works fine with azurite.

For this test, we want to use a user identity (Service Principal or an actual Azure AD user). The docs for `BlobServiceClient.get_user_delegation_key` say that

> A token credential must be present on the service object for this request to succeed.

Which seems to be incompatible with azurite. IIUC, it only supports authenticating with an account key or SAS token.